app/services/api/declarations/create.rb

```diff
@@ -0,0 +1,191 @@
+    validates :declaration_date,
+              declaration_date_within_milestone: true,
+              api_date_time_format: true,
+              allow_blank: true
+    validates :declaration_type, presence: { message: "Enter a '#/declaration_type'." }
+    validates :evidence_type, evidence_type: true
+    validate :validate_only_started_or_completed_if_mentor
+    validate :teacher_not_withdrawn_before_declaration_date
+    validate :validate_milestone_exists
+    validate :validates_billable_slot_available
+    validate :payment_statement_available
+
+    def create
+      return false unless valid?
+
+      Declarations::Create.new(
+        author:,
+        lead_provider:,
+        teacher:,
+        training_period:,
+        declaration_date:,
+        declaration_type:,
+        evidence_type: EvidenceTypeValidator.evidence_type_allowed?(self) ? evidence_type : nil,
+        payment_statement:,
+        mentorship_period:,
+        delivery_partner:
+      ).create
+    end
+
+    def training_period
+      return unless teacher
+
+      @training_period ||= case teacher_type
+                           when :ect
+                             teacher.ect_training_periods
+                             .includes(:lead_provider)
+                             .where(active_lead_providers: { lead_provider_id: })
+                             .closest_to(declaration_date)
+                             .first
+                           when :mentor
+                             teacher.mentor_training_periods
+                             .includes(:lead_provider)
+                             .where(active_lead_providers: { lead_provider_id: })
+                             .closest_to(declaration_date)
+                             .first
+                           end
+    end
+
+    def milestone
+      @milestone ||= schedule&.milestones&.find_by(declaration_type:)
+    end
+
+  private
+
+    def teacher
+      @teacher ||= Teacher.find_by(api_id: teacher_api_id) if teacher_api_id
+    end
+
+    def schedule
+      @schedule ||= training_period&.schedule
+    end
+
+    def contract_period
+      @contract_period ||= schedule&.contract_period
+    end
+
+    def mentorship_period
+      return unless training_period.for_ect?
+
+      @mentorship_period ||= training_period.trainee.mentorship_periods.closest_to(declaration_date).first
+    end
+
+    def delivery_partner
+      @delivery_partner ||= training_period.delivery_partner
+    end
+
+    def training_status
+      @training_status ||= API::TrainingPeriods::TrainingStatus.new(training_period:) if training_period
+    end
+
+    def existing_declarations
+      @existing_declarations ||= if training_period.for_ect?
+                                   teacher.ect_declarations
+                                 else
+                                   teacher.mentor_declarations
+                                 end
+    end
+
+    def existing_declaration
```

Comment:
Can we distinguish this from `existing_declarations`, maybe something like `existing_duplicate_declaration`?